sparql/queries/interactive-complex-1.sparql

```diff
@@ -29,10 +29,12 @@ SELECT
         WHERE
         {
             ?fr a snvoc:Person . 
-            ?fr snvoc:firstName "John" . 
+            ?fr snvoc:firstName $firstName .
             ?fr snvoc:id ?frId .
-            sn:pers00000000000000000933 snvoc:id ?rootId .
-            sn:pers00000000000000000933 ((snvoc:knows/snvoc:hasPerson)|^(snvoc:knows/snvoc:hasPerson))?/((snvoc:knows/snvoc:hasPerson)|^(snvoc:knows/snvoc:hasPerson))?/((snvoc:knows/snvoc:hasPerson)|^(snvoc:knows/snvoc:hasPerson)) ?fr .
+            BIND( "$personId"^^xsd:long AS ?rootId )
+            ?rootPerson a snvoc:Person .
+            ?rootPerson snvoc:id ?rootId .
+            ?rootPerson ((snvoc:knows/snvoc:hasPerson)|^(snvoc:knows/snvoc:hasPerson))?/((snvoc:knows/snvoc:hasPerson)|^(snvoc:knows/snvoc:hasPerson))?/((snvoc:knows/snvoc:hasPerson)|^(snvoc:knows/snvoc:hasPerson)) ?fr .
             FILTER(?frId != ?rootId)
```

Comment:
Maybe this can be simplified to `FILTER(?fr != ?rootPerson)`.
If so, we can ditch the whole `BIND(... AS ?rootId)` part.